RadarSDK/RadarAddress.m

```diff
@@ -0,0 +1,186 @@
+//
+//  RadarAddress.m
+//  RadarSDK
+//
+//  Copyright © 2019 Radar Labs, Inc. All rights reserved.
+//
+
+#import "RadarAddress+Internal.h"
+#import "RadarCoordinate+Internal.h"
+
+@implementation RadarAddress
+
++ (NSArray<RadarAddress *> * _Nullable)addressesFromObject:(id _Nonnull)object {
+    if (!object || ![object isKindOfClass:[NSArray class]]) {
+        return nil;
+    }
+
+    NSArray *addressesArr = (NSArray *)object;
+    NSMutableArray<RadarAddress *> *mutableAddresses = [NSMutableArray<RadarAddress *> new];
+
+    for (id addressObj in addressesArr) {
+        RadarAddress *address = [[RadarAddress alloc] initWithObject:addressObj];
+        if (!address) {
+            return nil;
+        }
+        [mutableAddresses addObject:address];
+    }
+
+    return mutableAddresses;
+}
+
+- (instancetype _Nullable)initWithObject:(id _Nonnull)object {
+    if (!object || ![object isKindOfClass:[NSDictionary class]]) {
+        return nil;
+    }
+
+    NSDictionary *addressDict = (NSDictionary *)object;
+
+    NSNumber *latitude;
+    NSNumber *longitude;
+    RadarCoordinate *coordinate;
+
+    NSString *formattedAddress;
+    NSString *country;
+    NSString *countryCode;
+    NSString *countryFlag;
+    NSString *state;
+    NSString *stateCode;
+    NSString *postalCode;
+    NSString *city;
+    NSString *borough;
+    NSString *county;
+    NSString *neighborhood;
+    NSString *number;
+
+    RadarAddressConfidence confidence = RadarAddressConfidenceNone;
```

Comment:
nit: we could move this down by the confidence parsing code

Reply:
right, definitely would make things easier to read -- was getting compiler warnings that `confidence` is potentially uninitialized if I wasn't assigning a value here. Could happen for example if the SDK gets something like an array back for `confidence` and it doesn't enter the block to map it to the enum value.

I should mark the enum entry for `None = 0` so we explicitly set it as the default regardless.

Reply:
there is precedent for initializing the "Unknown / None" enum variety in our other deserializers - https://github.com/radarlabs/radar-sdk-ios/blob/v3/RadarSDK/RadarEvent.m#L70